### PR TITLE
PHP7 support

### PR DESCRIPTION
**Pull request by DavertMik**
Actually everything worked for PHP7 with no changes. This PR enables testing for PHP7 and adds syntax checking for loaded tests. For PHP < 7 we use simple "php -l" linter.

### CODE REVIEW

**Comment by DavertMik** [comment]
squashed the commits. Please review

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,8 @@
 
 #### 2.1.5
 
+* **PHP7 support** verified )
```
Comment (samdark) [inline CHANGELOG.md]: `verified )`?

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,8 @@
...
+* Syntax check for tests. If PHP7 is used, `ParseException` handles syntax error, otherwise linting happens with `php -l`. HHVM - check is skipped. @davertmik
```
Comment (samdark) [inline CHANGELOG.md]: Why? HHVM should support `php -l` as well.

File: src/Codeception/Lib/Parser.php [hunk 3]
```diff
@@ -125,9 +124,33 @@ protected function addCommentStep($comment)
         $this->scenario->addStep(new \Codeception\Step\Comment($comment, []));
     }
 
+    public static function validate($file)
+    {
+        if (defined('HHVM_VERSION') // not for HHVM
+            || (strtoupper(substr(PHP_OS, 0, 3)) === 'WIN') // not for WIN
```
Comment (samdark) [inline src/Codeception/Lib/Parser.php]: `php -l` works well under Windows.

File: tests/unit/Codeception/Lib/ParserTest.php [hunk 4]
```diff
@@ -150,4 +150,43 @@ public function testParseFileWhichUnsetsFileVariable()
...
     }
-}
\ No newline at end of file
+    /**
+     * @group core
+     * @throws \Codeception\Exception\TestParseException
+     */
+    public function testCeptValidation()
+    {
+        if (defined('HHVM_VERSION') // not for HHVM
```
Comment (samdark) [inline tests/unit/Codeception/Lib/ParserTest.php]: It should work both with HHVM and Windows. `php -l` is supported in both cases.

File: CHANGELOG.md (hunk 2 above)
Comment (DavertMik) [inline CHANGELOG.md]: I was not sure about this... Is that really so?

File: src/Codeception/Lib/Parser.php (hunk 3 above)
Comment (DavertMik) [inline src/Codeception/Lib/Parser.php]: Yes, it works but Appveyor fails there. Somehow it doesn't handle well the error from another process
https://ci.appveyor.com/project/DavertMik/codeception/build/1.0.1020#L504

File: CHANGELOG.md (hunk 2 above)
Comment (samdark) [inline CHANGELOG.md]: Yes according to docs. Travis will show you.

Comment (DavertMik) [inline CHANGELOG.md]: Ok, HHVM is not that evil )

File: src/Codeception/Lib/Parser.php (hunk 3 above)
Comment (samdark) [inline src/Codeception/Lib/Parser.php]: https://stackoverflow.com/questions/9114574/php-exec-does-not-return-output

**Comment by DavertMik** [comment]
Enabled linting for HHVM as suggested by @samdark 
If no objections - will megre that in next hours

**Comment by samdark** [comment]
It's still unclear what https://github.com/Codeception/Codeception/pull/2590/files#r46268974 means.

**Comment by samdark** [comment]
Docs here https://github.com/Codeception/Codeception/pull/2590/files#r46269016 should be adjusted.

**Comment by samdark** [comment]
And... how about Windows?

**Comment by DavertMik** [comment]
For Windows we have 2 options:

* skip the appveyor test if it is failing
* skip windows validation

If we will have linter issues (like "php -l" not getting executed) that probably will make Codeception unusable on Windows. If we just ignore linting for windows it will work as it is now...

**Comment by samdark** [comment]
@DavertMik it's possible to solve linting on Windows: https://stackoverflow.com/questions/9114574/php-exec-does-not-return-output

**Comment by firodj** [comment]
@DavertMik We are using HHVM 3.9.1 which are defined as `PHP_MAJOR_VERSION=5`.
As our tests is growing, the start up time when running tests also increasing significantly.
After investigated we found that the lint validation took up about ~120 ms per test file, and 4 minutes total for all files just for linting.

Can this validation as optionally disabled?

**Comment by Naktibalda** [comment]
@firodj Could you raise it as a separate issue?

**Pull request merged**